### PR TITLE
feat: introduce my account schematics

### LINKED ISSUE #9029: Create @spartacus/myaccount schematis
We should start moving feature-specific schematics to the corresponding libraries, and the first candidate for that is the myaccount library.

### PR DESCRIPTION
**Pull request by znikola**
closes #9029

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,10 +90,11 @@
     "build:cds:ssr": "cross-env SPARTACUS_CDS=true ng run storefrontapp:server:production",
     "prerender:cds": "cross-env SPARTACUS_CDS=true ng run storefrontapp:prerender",
     "release:cds:with-changelog": "cd projects/cds && release-it && cd ../..",
+    "build:myaccount:schematics": "./feature-libs/my-account/build-schematics.sh",
     "build:myaccount": "ng build my-account --prod",
     "start:b2b": "cross-env SPARTACUS_BASE_URL=https://localhost:9002 SPARTACUS_API_PREFIX=/occ/v2/ SPARTACUS_B2B=true ng serve --prod",
     "build:product": "ng build product --prod",
-    "release:myaccount:with-changelog": "cd feature-libs/my-account && release-it && cd ../..",
+    "release:myaccount:with-changelog": "cd feature-libs/my-account && yarn build:myaccount:schematics && release-it && cd ../..",
```
Comment (marlass) [inline package.json]: Could you move that command to `release-it` after bump step?

File: feature-libs/my-account/schematics/add-my-account/index.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  ProjectType,
+  WorkspaceProject,
+  WorkspaceSchema,
+  WorkspaceTargets,
+} from '@schematics/angular/utility/workspace-models';
+import * as ts from 'typescript';
+import { version } from '../../package.json';
+import { Schema as MyAccountOptions } from './schema';
+
+export function addSpartacus(options: MyAccountOptions): Rule {
```
Comment (marlass) [inline feature-libs/my-account/schematics/add-my-account/index.ts]:
```suggestion
export function addSpartacusMyAccount(options: MyAccountOptions): Rule {
```

File: package.json [hunk 3]
```diff
@@ -90,7 +90,10 @@
     "build:cds:ssr": "cross-env SPARTACUS_CDS=true ng run storefrontapp:server:production",
     "prerender:cds": "cross-env SPARTACUS_CDS=true ng run storefrontapp:prerender",
     "release:cds:with-changelog": "cd projects/cds && release-it && cd ../..",
-    "build:myaccount": "ng build my-account --prod",
+    "cleanup:myaccount:schematics": "rimraf \"feature-libs/my-account/schematics/**/*.js\" \"feature-libs/my-account/schematics/**/*.js.map\" \"feature-libs/my-account/schematics/**/*.d.ts\"",
```
Comment (marlass) [inline package.json]: Can we move this command and the build one to my-account package.json and only reference it here (the build command, cleanup only there)?

File: package.json (hunk 1 above)
Comment (znikola) [reply package.json]: Done

File: package.json (hunk 3 above)
Comment (znikola) [reply package.json]: Done

File: feature-libs/my-account/.release-it.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
     "publishPath": "./../../dist/my-account"
   },
   "hooks": {
-    "after:version:bump": "cd ../.. && ng build my-account --prod"
+    "after:version:bump": "cd ../.. && ng build my-account --prod && yarn build:schematics"
```
Comment (marlass) [inline feature-libs/my-account/.release-it.json]:
```suggestion
    "after:version:bump": "cd ../.. && ng build my-account --prod && cd feature-libs/my-account && yarn build:schematics"
```

Comment (marlass) [reply feature-libs/my-account/.release-it.json]: Maybe this version is even better

```suggestion
    "after:version:bump": "cd ../.. && yarn build:myaccount"
```

File: feature-libs/my-account/package.json [hunk 5]
```diff
@@ -9,18 +9,34 @@
     "storefront",
     "my account"
   ],
+  "scripts": {
+    "clean:schematics": "yarn && rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
```
Comment (marlass) [inline feature-libs/my-account/package.json]:
```suggestion
    "clean:schematics": "rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
```

Comment (marlass) [reply feature-libs/my-account/package.json]: Running yarn on every test run is a bit of overkill ;)

**Review by marlass — CHANGES_REQUESTED** [review]

File: feature-libs/my-account/package.json (hunk 5 above)
Comment (znikola) [reply feature-libs/my-account/package.json]: It's required to install the `rimraf` dep. I can try to run `yarn` in the ci script

**Review by marlass — APPROVED** [review]

**Pull request merged**